src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3034,6 +3045,124 @@ public static void EnsureResizeSucceeds()
             }
         }
 
+        [Fact]
+        public static void ValueEquals_Null_True()
+        {
+            const string jsonString = "   null   ";
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+
+            Assert.True(jElement.ValueEquals((string)null));
+            Assert.True(jElement.ValueEquals(default(ReadOnlySpan<char>)));
+
+            Assert.False(jElement.ValueEquals((ReadOnlySpan<byte>)null));
+            Assert.False(jElement.ValueEquals(default(ReadOnlySpan<byte>)));
+        }
+
+        [Fact]
+        public static void ValueEquals_EmptyJsonString_True()
+        {
+            const string jsonString = "\"\"";
+            JsonElement jElement = (JsonElement)JsonDocument.Parse(jsonString).RootElement.Clone();
+
+            Assert.True(jElement.ValueEquals(""));
+            Assert.True(jElement.ValueEquals("".AsSpan()));
+            Assert.True(jElement.ValueEquals(default(ReadOnlySpan<char>)));
+            Assert.True(jElement.ValueEquals(default(ReadOnlySpan<byte>)));
+        }
+
+        [Fact]
+        public static void ValueEquals_TestTextEqualsLargeMatch_True()
+        {
+            var jsonChars = new char[320];  // Some value larger than 256 (stack threshold)
+            jsonChars.AsSpan().Fill('a');
+
+            Span<char> lookupChars = new char[jsonChars.Length];
+            jsonChars.CopyTo(lookupChars);
```

Comment:
This seems weird.  Why not just

```C#
string lookup = new string('a', 320);
string jsonString = "\"" + lookup + "\"";

using (JsonDocument doc = JsonDocument.Parse(jsonString))
{
   ...
}
```

That is, the two manual arrays and one string can all just be one string.